xdsl/dialects/memref.py

```diff
@@ -147,15 +149,29 @@ class Alloc(IRDLOperation):
 
     irdl_options = [AttrSizedOperandSegments(as_property=True)]
 
-    @staticmethod
+    def __init__(
+        self,
+        dynamic_sizes: Sequence[SSAValue],
+        symbol_operands: Sequence[SSAValue],
+        result_type: Attribute,
+        alignment: Attribute | None = None,
+    ):
+        super().__init__(
+            operands=(dynamic_sizes, symbol_operands),
+            result_types=(result_type,),
+            properties={"alignment": alignment},
+        )
+
+    @classmethod
```

Comment:
can deprecate this in the future

Reply:
Dumb question: Why is this suddenly a classmethod?

Reply:
That was a remainder of a WIP change, I can change it back now, I just felt like it didn't hurt.